### PR TITLE
Shopping cart example

### PR DESCRIPTION
**Pull request by hedgerh**
For some reason the product images can't be found.  Guessing it is some webpack config issue, if anyone has any suggestions.

Other than that, it should be good to go.

As per issue https://github.com/rackt/redux/issues/823

### CODE REVIEW

**Comment by hedgerh** [comment]
@gaearon

**Comment by hedgerh** [comment]
Going to clean up the linting errors later tonight.  Still not sure about why product images are 404ing.

**Comment by gaearon** [comment]
>For some reason the product images can't be found.

No need for images—let's keep it simple.

File: examples/shopping-cart/components/Cart.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      } - & euro; {
+        p.price
+      }
+      x {
+        p.quantity
+      } < /Product>;
+    });
+
+    return ( < div className = "cart uk-panel uk-panel-box uk-panel-box-primary" >
+      < div className = "uk-badge uk-margin-bottom" > Your Cart < /div> < div className = "uk-margin-small-bottom" > {
```
Comment (npverni) [inline examples/shopping-cart/components/Cart.js]: Is there a reason for the space before the tag names here?

Comment (hedgerh) [inline examples/shopping-cart/components/Cart.js]: @npverni Weird side effect of the prettifying tool I used to convert from 4 space -> 2 space indentation.  Fixed!  Still need to work on the linting issues.

**Comment by hedgerh** [comment]
This is ready for review. :octopus:

File: examples/shopping-cart/actions/index.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    dispatch({
+      type: types.CHECKOUT_REQUEST
+    });
+    shop.buyProducts(products, () => {
+      dispatch({
+        type: types.CHECKOUT_SUCCESS,
+        cart
+      });
+      // Replace the line above with line below to rollback on failure:
+      // dispatch({ type: CHECKOUT_FAILURE, cart });
```
Comment (omnidan) [inline examples/shopping-cart/actions/index.js]: in the comment, this should be `types.CHECKOUT_FAILURE`

**Comment by omnidan** [comment]
@hedgerh small typo in the comment, otherwise it looks good to me :ok_hand:

File: examples/shopping-cart/components/Cart.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+export default class Cart extends Component {
+  render() {
+    const { products, total, onCheckoutClicked } = this.props;
+
+    const hasProducts = products.length > 0;
+    const nodes = !hasProducts ?
+      <div>Please add some products to cart.</div> :
+      products.map(product =>
+        <Product key={product.id}>
+          {product.title} - &euro;{product.price} x {product.quantity}
```
Comment (omnidan) [inline examples/shopping-cart/components/Cart.js]: Any reason why you don't pass the title, price and quantity to `<Product>`?

File: examples/shopping-cart/components/ProductItem.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import React, { Component, PropTypes } from 'react';
+
+export default class ProductItem extends Component {
+  render() {
+    const { product } = this.props;
+
+    return (
+      <div>
+        <h4>{product.title} - &euro;{product.price}</h4>
```
Comment (omnidan) [inline examples/shopping-cart/components/ProductItem.js]: If you do what I said above, you could re-use `<Product>` here

**Comment by hedgerh** [comment]
Thanks for taking the time to look it over, Daniel. And nice catch! I am
too full of food to get out of bed, but I'll fix it tomorrow :D

**Comment by hedgerh** [comment]
@omnidan: Made the corrections to the comment typo and I'm passing the title/quantity/price to the Product component as props.  Also changed ProductItem to use the Product component.  Added a little check in Product's render method to not render the 'x' if quantity wasn't passed.

**Comment by omnidan** [comment]
@hedgerh looks good now :+1:

**Comment by gaearon** [comment]
I'd appreciate a few fixes:

**Items need paddings.**

Right now they're way to close.

screen shot 2015-10-28 at 15 11 22

**No need to mention Flux :-)**

Let's change “Flux Shop Demo (Redux)” title to “Shopping Cart”.

**Pull request merged**